packages/shared/test-suites/RecordingTestMetadata.ts

```diff
@@ -350,6 +350,21 @@ export async function processCypressTestRecording(
 
         for (let index = 0; index < annotations.length; index++) {
           const annotation = annotations[index];
+
+          if (!beginPoint || comparePoints(beginPoint.point, annotation.point) > 0) {
+            beginPoint = {
+              point: annotation.point,
+              time: annotation.time,
+            };
+          }
+
+          if (!endPoint || comparePoints(endPoint.point, annotation.point) < 0) {
+            endPoint = {
+              point: annotation.point,
+              time: annotation.time,
+            };
+          }
```

Comment:
This looks like a bit of a hack around bad data. Do we need to do this?

If we do, I'd love if we could add a follow up SCS issue (with a TODO comment referencing it) to remove it later.

Reply:
I think this may also be a bit flaky. If annotations are sorted by execution points, and we hit the `! beginPoint` case, when when we later hit the `"test:start"` case below, we would erase the `beginPoint` we'd set here.

```js
            case "test:start": {
              beginPoint = {
                point: annotation.point,
                time: annotation.time,
              };
              break;
            }
```

To handle that we'd need to also have a `comparePoints` check in `"test:start"`.

Reply:
This is the crux of the problem. `test:start` and `test:end` annotations are not reliable at this time because the plugin cannot guarantee it runs before and after all other commands. I'll remove the `test:start` and `test:end` handling altogether to avoid confusion.

Reply:
I'm just not clear why the plug-in can't do a similar type of detection/logic as the frontend is doing here. It collets these annotations, and (eventually ) saves them right? Why can't it do a check _before_ saving them.

I guess I'm concerned that our annotation data has known unreliable cases like this.

Reply:
> I'll remove the test:start and test:end handling altogether to avoid confusion.

That does seem like it would be an improvement, since (I think) as things currently are– it would break.

Reply:
The annotations are created as events happen in the browser. They're not collected and applied later so it's challenging to know that a new test has started and more events from a previous test aren't still going to come in.

Reply:
Ah, I see. I'm not very knowledgeable about annotations stuff. I assumed there was a post-processing step, but I guess that's only true if we run a routine and generate derived annotations?

Reply:
Perhaps? I'm not very knowledgeable about routine stuff :)

Reply:
I think the high level way it works is that some type of annotation(s) get recorded while the recording is being made, and then a routine runs and loops over them, and stores derived annotations.

In our case, I guess that would mean the raw test+step events (and maybe navigation events) could get pre-processed and normalized before the frontend saw them.

That might be worth considering at some point anyway, for smoothing over differences between Cypress and Playwright. (Not sure how far @hbenl has thought through the Playwright side of this yet. I haven't really.)